mriqc/instrumentation/resources.py

```diff
@@ -144,17 +171,10 @@ def __init__(
 
     def run(self, *args, **kwargs):
         """Core monitoring function, called by start()"""
-
+        print("run")
         # Open file now, because it cannot be pickled.
-        _logfile = (
-            Path(self._logfile)
-            if self._logfile
-            else Path(mkstemp(prefix="prof-", suffix=".tsv")[1])
-        )
-        _logfile.parent.mkdir(parents=True, exist_ok=True)
-        self._logfile = str(_logfile.absolute())
-
-        _logfile = _logfile.open("w")
+        _logfile = open(self._logfile, "w")
```

Comment:
Please keep the old code unless something specific is being fixed. If so, please leave a comment saying what is the problem.
```suggestion
        _logfile = (
            Path(self._logfile)
            if self._logfile
            else Path(mkstemp(prefix="prof-", suffix=".tsv")[1])
        )
        _logfile.parent.mkdir(parents=True, exist_ok=True)
        self._logfile = str(_logfile.absolute())

        _logfile = _logfile.open("w")
```

Reply:
I re-added the import statement for mkstemp, now it works with the old code.